core/src/main/scala/com/softwaremill/sttp/model/ResponseAs.scala

```diff
@@ -39,3 +41,9 @@ case class MappedResponseAs[T, T2, S](raw: ResponseAs[T, S], g: T => T2)
   override def map[T3](f: T2 => T3): ResponseAs[T3, S] =
     MappedResponseAs[T, T3, S](raw, g andThen f)
 }
+
+case class ResponseAsFile(file: File, overwrite: Boolean)
```

Comment:
isn't it enough to have just one, and convert to `File`/`Path` in `responseAsFile` or `responseAsPath`?